esp/xslt/esdl2monitor.xslt

```diff
@@ -0,0 +1,976 @@
+    export integer1 OuterLeft := 1;
+    export integer1 OuterRight := 2;
+  END;
+
+  // Records in the old and new results are matched by virtual record ID (VID) -- something,
+  // which makes "this" record unique. Usually is defined by business logic.
+  // If two records have the same VID, they are compared field by field.
+  EXPORT State := MODULE
+    export integer1 VOID           := 0; // no comparison occured (for example, no alerts were requested)
+    export integer1 UNCHANGED      := 1; // no changes
+    export integer1 UPDATED        := 2; // there are changes to a scalar field
+    export integer1 ADDED          := 4; // record is new (with respect to VID)
+    export integer1 DELETED        := 8; // record doesn't exist anymore
+    export integer1 PREVIOUS      := 16; // for UPDATED records only: will only have populated fields which are changed
+    export integer1 CHILD_UPDATED := 32; // a record has a child dataset which has some records ADDED, DELETED or UPDATED
+  END;
+
+  EXPORT string Convert (integer sts) :=
+    MAP (sts = State.DELETED   => 'deleted',
+         sts = State.ADDED     => 'added',
+         sts = State.UPDATED   => 'updated',
+         sts = State.UNCHANGED => '',
+         '');
+END;
+
+layouts := MODULE
+
+  EXPORT DiffMetaRow := RECORD
+    string name {XPATH ('@name')};
+    string prior {XPATH ('@prior')};
+  END;
+
+  EXPORT DiffMetaRec := RECORD
+    string7 _child_diff {XPATH('@child_diff')} := '';
+    string7 _diff {XPATH('@diff')} := '';
+    DATASET (DiffMetaRow) _diffmeta {XPATH ('DiffMeta/Field')} := DATASET ([], DiffMetaRow);
+  END;
+
+  <xsl:apply-templates select="EsdlStruct" mode="layouts"/>
+  <xsl:apply-templates select="EsdlRequest" mode="layouts"/>
+  <xsl:apply-templates select="EsdlResponse" mode="layouts"/>
+  <xsl:text>END;
+
+</xsl:text>
+difference := MODULE
+    <xsl:for-each select="Selectors/Selector">
+      shared boolean Monitor<xsl:value-of select="."/> := FALSE : STORED('Monitor_<xsl:value-of select="."/>', FORMAT(sequence(<xsl:value-of select="position()+10"/>)));
+    </xsl:for-each>
+<xsl:text>
+  shared set of string optional_fields := [
+</xsl:text>
+    <xsl:for-each select="OptionalFields/Path">
+      <xsl:if test="position()!=1">,</xsl:if><xsl:text>'/</xsl:text><xsl:value-of select="."/><xsl:text>'
+    </xsl:text>
+    </xsl:for-each>
+       <xsl:text>
+    ];
+</xsl:text>
+       <xsl:apply-templates select="EsdlStruct" mode="difference.struct"/>
+       <xsl:apply-templates select="EsdlRequest" mode="difference.struct"/>
+       <xsl:apply-templates select="EsdlResponse" mode="difference.struct"/>
+<xsl:text>END;
+
+</xsl:text>
+  the_differenceModule := difference._df_<xsl:value-of select="$responseType"/>;
+  the_requestLayout := layouts._lt_<xsl:value-of select="$requestType"/>;
+
+  the_responseLayout := layouts._lt_<xsl:value-of select="$responseType"/>;
+
+
+
+default_service_url := 'http://roxiestaging.sc.seisint.com:9876/';
```

Comment:
this looks intentional, but felt it was worth pointing out in case we didn't mean to hardcode an internal url